internal/pkg/dashboard/dashboard.go

```diff
@@ -156,27 +166,31 @@ func New(cli *client.Client, opts ...Option) (*Dashboard, error) {
 		return t.screenKey, string(t.screen)
 	})
 
-	screenKeyCodeToScreen := slices.ToMap(dashboard.screenConfigs, func(t screenConfig) (tcell.Key, Screen) {
-		return t.keyCode, t.screen
+	screenConfigByKeyCode := slices.ToMap(dashboard.screenConfigs, func(config screenConfig) (tcell.Key, screenConfig) {
+		return config.keyCode, config
 	})
 
 	dashboard.footer = components.NewFooter(screenKeyToName)
 
 	dashboard.app.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
-		screenName, screenOk := screenKeyCodeToScreen[event.Key()]
+		config, screenOk := screenConfigByKeyCode[event.Key()]
+
+		allowNodeNavigation := dashboard.selectedScreenConfig != nil && dashboard.selectedScreenConfig.allowNodeNavigation
 
 		switch {
 		case screenOk:
-			dashboard.selectScreen(screenName)
-		case event.Key() == tcell.KeyLeft, event.Rune() == 'h':
+			dashboard.selectScreen(config.screen)
+
+			return nil
+		case allowNodeNavigation && event.Key() == tcell.KeyLeft || event.Rune() == 'h':
```

Comment:
If I understand this right you want this:
```go
    allowNodeNavigation && (event.Key() == tcell.KeyLeft || event.Rune() == 'h')
``` 

but currently it is something like this:

```go
(allowNodeNavigation && event.Key() == tcell.KeyLeft) || event.Rune() == 'h'
```

https://go.dev/play/p/eW0z8yrDqkx

Reply:
yes exactly, missed this one 🤦 I know boolean logic 😆 fixed